tests/check/elements/rtpjitterbuffer.c

```diff
@@ -3684,6 +3686,109 @@ GST_START_TEST (test_dtx_without_clock)
 
 GST_END_TEST;
 
+GST_START_TEST (test_ulpfec_large_pkt_spacing)
+{
+  gint latency_ms = 20;
+  gint frame_dur_ms = 50;
+  gint i, seq;
+  GstBuffer *buffer;
+  GstClockTime now;
+  GstClockTime frame_dur = frame_dur_ms * GST_MSECOND;
+  GstHarness *h = gst_harness_new ("rtpjitterbuffer");
+
+  g_object_set (h->element, "do-lost", TRUE, "latency", latency_ms, NULL);
+  gst_harness_set_src_caps (h, generate_caps ());
+
+  /* Pushing 2 frames @frame_dur_ms ms apart from each other to initialize
+   * packet_spacing and avg jitter */
+  for (seq = 0, now = 0; seq < 2; ++seq, now += frame_dur) {
+    gst_harness_set_time (h, now);
+    gst_harness_push (h, generate_test_buffer_full (now, seq,
+            AS_TEST_BUF_RTP_TIME (now)));
+    if (seq == 0)
+      gst_harness_crank_single_clock_wait (h);
+    buffer = gst_harness_pull (h);
+    fail_unless_equals_int64 (now, GST_BUFFER_PTS (buffer));
+    gst_buffer_unref (buffer);
+  }
+
+  /* drop GstEventStreamStart & GstEventCaps & GstEventSegment & Latency */
+  for (i = 0; i < 4; i++)
+    gst_event_unref (gst_harness_pull_event (h));
+  /* drop reconfigure event */
+  gst_event_unref (gst_harness_pull_upstream_event (h));
+
+  /* Pushing packet #2 as ULPFEC */
+  now = seq * frame_dur;
+  gst_harness_set_time (h, now);
+  buffer = generate_test_buffer_full (now, seq, AS_TEST_BUF_RTP_TIME (now));
+  GST_BUFFER_FLAG_SET (buffer, GST_RTP_BUFFER_FLAG_ULPFEC);
+  fail_unless_equals_int (GST_FLOW_OK, gst_harness_push (h, buffer));
+  buffer = gst_harness_pull (h);
+  fail_unless_equals_int64 (now, GST_BUFFER_PTS (buffer));
+  gst_buffer_unref (buffer);
+  fail_unless_equals_int (0, gst_harness_buffers_in_queue (h));
+
+
+  /* Packet #3 should have PTS not affected by clock skew logic */
+  seq += 1;
+  now = seq * frame_dur;
+  gst_harness_set_time (h, now);
+  gst_harness_push (h, generate_test_buffer_full (now, seq,
+          AS_TEST_BUF_RTP_TIME (now)));
+  buffer = gst_harness_pull (h);
+  fail_unless_equals_int64 (now, GST_BUFFER_PTS (buffer));
+  gst_buffer_unref (buffer);
+
+  gst_harness_teardown (h);
+}
+
+GST_END_TEST;
+
+
+static const JitterBufferSkewCtx
+    ulpfec_pkt_does_not_affect_pts_calculation_input[] = {
+  {0, 0},
+  {20 * GST_MSECOND, -100},
+  {-10 * GST_MSECOND, 1},
+  {100 * GST_MSECOND, 0},
+};
+
+GST_START_TEST (test_ulpfec_does_not_affect_pts_calculation)
```

Comment:
I had one suggestion here. If these tests are more of less identical for RTX and ULPFEC, how about a enum or something inside JitterBufferSkewCtx so that the test can do a variation of RTX and ULPFEC, since they are treated pretty much the same now?

Reply:
Yeah I thought about passing the buffer flag that is set into a common test function, I believe that will work :).